app/src/main/java/tech/relaycorp/gateway/domain/publicsync/CollectParcelsFromGateway.kt

```diff
@@ -102,8 +101,10 @@ class CollectParcelsFromGateway
                 logger.info("Parcel already received")
             is StoreParcel.Result.Success -> {
                 logger.info("Collected parcel from Gateway ${storeResult.parcel.id}")
-                if (keepAlive) {
-                    notifyEndpoints.notify(MessageAddress.of(storeResult.parcel.recipientAddress))
+                if (keepAlive && storeResult.parcel.recipient.internetAddress != null) {
+                    notifyEndpoints.notify(
+                        MessageAddress.of(storeResult.parcel.recipient.internetAddress!!)
+                    )
```

Comment:
The `if` condition shouldn't have changed, and we should just replace `storeResult.parcel.recipientAddress` with `storeResult.parcel.recipient.id`.

This code is meant to notify local apps about incoming parcels, so it wouldn't work if we use the Internet address as the index: all local apps share the same Internet gateway address.